src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs

```diff
@@ -65,6 +66,9 @@ public class LuceneIndexPersistence : IDisposable
 
         internal LuceneVoronDirectory LuceneDirectory => _directory;
 
+        private readonly object _readersLock = new object();
+        private readonly object _writersLock = new object();
```

Comment:
Why is this required? Writers are only created on the indexer thread, no?

Reply:
Answer is here - https://github.com/ravendb/ravendb/pull/11762/files#r579855989